<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Klara Kaleb" />

<meta name="date" content="2018-08-29" />

<title>Get started with OmaDB</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Get started with OmaDB</h1>
<h4 class="author"><em>Klara Kaleb</em></h4>
<h4 class="date"><em>2018-08-29</em></h4>



<p>This little vignette shows you how to get started with the <code>roma</code> package. roma is a wrapper for the REST API for the Orthologous MAtrix project (OMA) which is a database for the inference of orthologs among complete genomes.</p>
<p>For more details on the OMA project, see <a href="https://omabrowser.org/oma/home/" class="uri">https://omabrowser.org/oma/home/</a>.</p>
<div id="some-useful-functions" class="section level2">
<h2>Some useful functions</h2>
<p>The package contains a range of functions that are used to query the database in an R friendly way. This vignette describes some of them, whereas the rest are described in more detail in other vignettes:</p>
<p><a href="exploring_hogs.html">Exploring Hierarchical orthologous groups with roma</a></p>
<p><a href="tree_visualisation.html">Exploring Taxonomic trees with roma</a></p>
<p><a href="sequence_mapping.html">Sequence Analysis with roma</a></p>
<div id="getxref" class="section level3">
<h3>getXref</h3>
<p>This function searches the OMA database for entries containing the pattern defined and returns the results in a dataframe. Hence, it is usually a good starting place. Sample response is below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(OmaDB)

xref =<span class="st"> </span><span class="kw">load</span>(<span class="st">'../data/xref.rda'</span>)

<span class="kw">head</span>(xref)</code></pre></div>
<pre><code>## [1] &quot;xref&quot;</code></pre>
</div>
<div id="getgenomealignment" class="section level3">
<h3>getGenomeAlignment</h3>
<p>This function serves to obtain the orthologs for 2 whole genomes. The result is a dataframe containing information on each member in the pair and their relationship.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">'../data/pairs.rda'</span>)

<span class="kw">head</span>(pairs)</code></pre></div>
<pre><code>##   entry_1.entry_nr                           entry_1.entry_url
## 1          6618226 https://omabrowser.org/api/protein/6618226/
## 2          6618227 https://omabrowser.org/api/protein/6618227/
## 3          6618228 https://omabrowser.org/api/protein/6618228/
## 4          6618229 https://omabrowser.org/api/protein/6618229/
## 5          6618230 https://omabrowser.org/api/protein/6618230/
## 6          6618231 https://omabrowser.org/api/protein/6618231/
##   entry_1.omaid entry_1.canonicalid             entry_1.sequence_md5
## 1    ASHGO00001              Q75FB7 a9b1a6dc9afb2b02afe8fdf8029b5f22
## 2    ASHGO00002              Q75FB6 5d186037c4dd0a89b34d70d596fac86d
## 3    ASHGO00003              Q75FB5 3a83b276f0c9034f7cf66277e7d6c983
## 4    ASHGO00004              Q75FB4 a8611c3f24ac6599e6a36a2710f6d24a
## 5    ASHGO00005              Q75FB3 07277f0ca66fcb49d175667272f1547f
## 6    ASHGO00006              Q75FB2 2b666569856af5f068e532ade89c1140
##   entry_1.oma_group entry_1.oma_hog_id entry_1.chromosome
## 1                 0     HOG:0393392.4c                  I
## 2            203915        HOG:0200818                  I
## 3            214367        HOG:0200433                  I
## 4            768456  HOG:0387657.2d.3a                  I
## 5            530479     HOG:0397172.3b                  I
## 6            563083     HOG:0201049.3a                  I
##   entry_1.locus.start entry_1.locus.end entry_1.locus.strand
## 1                8108              9067                    1
## 2                9537             12593                    1
## 3               12906             13244                    1
## 4               13713             14846                    1
## 5               16155             19850                    1
## 6               20056             23721                   -1
##   entry_1.is_main_isoform entry_2.entry_nr
## 1                    TRUE          6637770
## 2                    TRUE          6637359
## 3                    TRUE          6637360
## 4                    TRUE          6637767
## 5                    TRUE          6636211
## 6                    TRUE          6636209
##                             entry_2.entry_url entry_2.omaid
## 1 https://omabrowser.org/api/protein/6637770/    YEAST04806
## 2 https://omabrowser.org/api/protein/6637359/    YEAST04395
## 3 https://omabrowser.org/api/protein/6637360/    YEAST04396
## 4 https://omabrowser.org/api/protein/6637767/    YEAST04803
## 5 https://omabrowser.org/api/protein/6636211/    YEAST03247
## 6 https://omabrowser.org/api/protein/6636209/    YEAST03245
##   entry_2.canonicalid             entry_2.sequence_md5 entry_2.oma_group
## 1          RCE1_YEAST 605098a0697ad8fc7af2101e758033cb            494558
## 2          ZDS2_YEAST 8cc75f16fbfd321833abc48fd1173154            203915
## 3          YMK8_YEAST 783fea3b573632292d89a5a5218b6e90            214367
## 4          SCS7_YEAST 9307c7a6e80ed39d8b6529329fee1819            768456
## 5          SMC3_YEAST 4e8f1295434b44ae8f749cad976b966e            530479
## 6          NET1_YEAST f2ba71aea520ea66f015ba357eb6e8c6            563083
##   entry_2.oma_hog_id entry_2.chromosome entry_2.locus.start
## 1     HOG:0393392.4c               XIII              814364
## 2     HOG:0200818.1a               XIII               51640
## 3        HOG:0200433               XIII               54793
## 4  HOG:0387657.2d.3a               XIII              809623
## 5     HOG:0397172.3b                  X              299157
## 6     HOG:0201049.2a                  X              295245
##   entry_2.locus.end entry_2.locus.strand entry_2.is_main_isoform rel_type
## 1            815311                   -1                    TRUE      1:1
## 2             54468                    1                    TRUE      1:1
## 3             55110                    1                    TRUE      1:1
## 4            810777                   -1                    TRUE      1:1
## 5            302849                   -1                    TRUE      1:1
## 6            298814                    1                    TRUE      1:1
##   distance   score
## 1 122.0000  636.04
## 2  95.0000 1424.24
## 3  50.0000  557.51
## 4  37.7442 2682.46
## 5  58.0000 5548.40
## 6  90.0000 1844.35</code></pre>
</div>
<div id="getdata" class="section level3">
<h3>getData</h3>
<p>This master-function function serves to obtain the information for a single entry in a database - either a group, protein or a genome. The data type is specified by setting the “type” argument and a specific entry by its ID - below are the possible ID’s for different object type.</p>
<ul>
<li>Protein
<ul>
<li>oma id</li>
<li>canonical id</li>
<li>entry number - release specific</li>
</ul></li>
<li>Genome
<ul>
<li>NCBI taxon id</li>
<li>UniProt species code</li>
</ul></li>
<li>Group
<ul>
<li>group number</li>
<li>group fingerprint</li>
<li>id of one of its members</li>
</ul></li>
</ul>
</div>
<div id="getobjectattributes" class="section level3">
<h3>getObjectAttributes</h3>
<p>The result of the getData function is an S3 Object with attributes corresponding to the information requested. This function allows the user to list all the object attributes and their corresponding data types.</p>
</div>
<div id="getattribute" class="section level3">
<h3>getAttribute</h3>
<p>The specific attributes of the created object can be accessed via $ or via the getAttribute() function. Below is an example of object containing information about an OMA group.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">'../data/group.rda'</span>)

object_attributes =<span class="st"> </span><span class="kw">getObjectAttributes</span>(group)</code></pre></div>
<pre><code>## [1] &quot;group_nr : integer&quot;
## [1] &quot;fingerprint : character&quot;
## [1] &quot;related_groups : URL&quot;
## [1] &quot;members : data.frame&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">group<span class="op">$</span>fingerprint</code></pre></div>
<pre><code>## [1] &quot;FPNDKFP&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getAttribute</span>(group, <span class="st">'fingerprint'</span>)</code></pre></div>
<pre><code>## [1] &quot;FPNDKFP&quot;</code></pre>
</div>
<div id="resolveurl" class="section level3">
<h3>resolveURL</h3>
<p>In most cases there is great quantity of information available for a given entry and this impacts the data retrival time. Due to this, the information available for such entries is split into a number of endpoints and these are included appropriatelly as redirects. This function allows the user to obtain further information behind those urls.</p>
<p>An example of use for the above function would be to obtain the list of orthologs for a given protein.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">'../data/protein.rda'</span>)

<span class="kw">getAttribute</span>(protein,<span class="st">'orthologs'</span>)</code></pre></div>
<pre><code>## [1] &quot;https://omabrowser.org/api/protein/6633022/orthologs/&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">'../data/orthologs.rda'</span>)

orthologs </code></pre></div>
<pre><code>##    entry_nr                                   entry_url      omaid
## 1   6342668 https://omabrowser.org/api/protein/6342668/ COCLU03588
## 2   6399341 https://omabrowser.org/api/protein/6399341/ COLGR10929
## 3   6407219 https://omabrowser.org/api/protein/6407219/ COLSU06788
## 4   6468764 https://omabrowser.org/api/protein/6468764/ FUSO411719
## 5   6475980 https://omabrowser.org/api/protein/6475980/ GIBZA02345
## 6   6501606 https://omabrowser.org/api/protein/6501606/ NECHA14651
## 7   6530466 https://omabrowser.org/api/protein/6530466/ THIHE03640
## 8   6554631 https://omabrowser.org/api/protein/6554631/ CANAL00775
## 9   6560817 https://omabrowser.org/api/protein/6560817/ CANAW00573
## 10  6566321 https://omabrowser.org/api/protein/6566321/ LODEL00382
## 11  6575886 https://omabrowser.org/api/protein/6575886/ DEBHA04175
## 12  6579264 https://omabrowser.org/api/protein/6579264/ PICGU01290
## 13  6594625 https://omabrowser.org/api/protein/6594625/ SPAPN04932
## 14  6595834 https://omabrowser.org/api/protein/6595834/ CANTE00168
## 15  6608129 https://omabrowser.org/api/protein/6608129/ KOMPG00513
## 16  6619568 https://omabrowser.org/api/protein/6619568/ ASHGO01343
## 17  6623679 https://omabrowser.org/api/protein/6623679/ KLULA00697
## 18  6630047 https://omabrowser.org/api/protein/6630047/ CANGA01835
## 19  6642012 https://omabrowser.org/api/protein/6642012/ ZYGRO02696
## 20  8837885 https://omabrowser.org/api/protein/8837885/ DROBM15099
##    canonicalid                     sequence_md5 oma_group     oma_hog_id
## 1              c5db7c8e6b0eee5c6bbc18fe2dbf9ba7    737617 HOG:0379998.2a
## 2              2cf82b1ffadc581d82d920af9b0130a2    737617 HOG:0379998.2a
## 3   A0A066XI05 d6abbfd6d6ed96217331e86733ae4aeb    737617 HOG:0379998.2a
## 4   A0A0D2XKA6 b92bedd8c4fe61cc3667d4f108ffb528    737617 HOG:0379998.2a
## 5              54814145e0f45beb33ac2e2cb99335a3    737618 HOG:0379998.2a
## 6              8c85e6f457288d146cb99fbc61404cb9    737617 HOG:0379998.2a
## 7              416de3ec8baf6df5b95a10b2145e7242    737617 HOG:0379998.2a
## 8   CCR4_CANAL b80e7f71d278fae2335d393c82912fee    737617 HOG:0379998.2b
## 9       C4YDK4 d1158bec0e2fface48020c27d9e6ca2b    737618 HOG:0379998.2b
## 10      A5DSP6 58726579b898e8bc18c954e8cc7039f5    737636 HOG:0379998.2b
## 11  CCR4_DEBHA 06ae2387578bfe3873ad4c34a339bb26    737636 HOG:0379998.2b
## 12      A5DDD9 d796187a45010e0868d774e568bf8d75    737617 HOG:0379998.2b
## 13      G3ATH1 fe2c181e7927a749ce89d63f0cfa99a9    737636 HOG:0379998.2b
## 14             4ff2469626a34edf6dfc0cc18d6ae06b    737618 HOG:0379998.2b
## 15      C4R821 eb4910adfeab0ff42a374bee5d9c2176    737618 HOG:0379998.2b
## 16  CCR4_ASHGO dc714e8db742c3fa6e16c074c3ea9b75    737617 HOG:0379998.2b
## 17  CCR4_KLULA bd38c48cb3b76bed0c72f5cd2a7ec92e    737636 HOG:0379998.2b
## 18  CCR4_CANGA 3329cf643b6767b0b022622779d5daf1    737636 HOG:0379998.2b
## 19  A0A1Q2ZZG0 f2629a0761ab52e0a1beb6b4a5d9c868    737636 HOG:0379998.2b
## 20             11d6a871f132d98e296ba5a628cac8b4         0               
##                                   chromosome locus.start locus.end
## 1                                 scaffold_6       27852     30189
## 2                                   GG697333      764822    767178
## 3                            Scaffolds0609.1         399      2755
## 4                                          4     4317479   4319279
## 5                            Supercontig_3.1     6701031   6703176
## 6                           sca_14_chr10_3_0      669785    671940
## 7                               chromosome_2     3756431   3758803
## 8                     supercontig_supercont4     1809654   1812017
## 9                                          1     1390917   1393274
## 10 supercont1.1 of Lodderomyces elongisporus     1013851   1016379
## 11                                         F      373608    376103
## 12                      supercontig_CH408156      187607    189769
## 13                                scaffold_6      636379    638832
## 14                            scaffold_00004      297506    299533
## 15                                         4      962805    965323
## 16                                       III      882342    884552
## 17                                         F     1468612   1470984
## 18                                         H      598643    601264
## 19                                         D      453024    455597
## 20                          scf7180000299460       79621     84056
##    locus.strand is_main_isoform rel_type distance   score
## 1            -1            TRUE      1:1 105.0000 1362.00
## 2             1            TRUE      1:1 108.0000 1320.99
## 3             1            TRUE      1:1 109.0000 1312.07
## 4            -1            TRUE      1:1 107.0000 1326.20
## 5            -1            TRUE      1:1 109.0000 1327.66
## 6            -1            TRUE      1:1 105.0000 1336.03
## 7             1            TRUE      1:1 113.0000 1251.50
## 8             1            TRUE      1:1  72.0000 2268.69
## 9            -1            TRUE      1:1  72.0000 2266.12
## 10           -1            TRUE      1:1  74.0000 2267.07
## 11           -1            TRUE      1:1  73.0000 2246.76
## 12           -1            TRUE      1:1  76.0000 2124.66
## 13           -1            TRUE      1:1  73.0000 2240.65
## 14            1            TRUE      1:1  79.0000 2116.79
## 15            1            TRUE      1:1  68.0000 2364.83
## 16            1            TRUE      1:1  36.1210 4302.76
## 17           -1            TRUE      1:1  36.1210 4423.53
## 18           -1            TRUE      1:1  35.3357 5201.71
## 19           -1            TRUE      1:1  42.1286 4655.84
## 20           -1            TRUE      1:1   0.3903 7730.37</code></pre>
<p>The orthologs for a protein are returned as a data.frame. This structure is also found in other areas of the package (e.g. the data on the members of a particular OMA group or a HOG) and hence features a function getInfo() to simplify its processing. For example, the user can obtain a set of genomic ranges for the proteins in a dataframe as so:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gRanges =<span class="st"> </span><span class="kw">getInfo</span>(orthologs,<span class="dt">type=</span><span class="st">'genomic_ranges'</span>)

<span class="kw">str</span>(gRanges)</code></pre></div>
<pre><code>## Formal class 'GRanges' [package &quot;GenomicRanges&quot;] with 7 slots
##   ..@ seqnames       :Formal class 'Rle' [package &quot;S4Vectors&quot;] with 4 slots
##   .. .. ..@ values         : Factor w/ 20 levels &quot;ZYGRO02696&quot;,&quot;ASHGO01343&quot;,..: 7 8 9 12 13 17 20 3 4 16 ...
##   .. .. ..@ lengths        : int [1:20] 1 1 1 1 1 1 1 1 1 1 ...
##   .. .. ..@ elementMetadata: NULL
##   .. .. ..@ metadata       : list()
##   ..@ ranges         :Formal class 'IRanges' [package &quot;IRanges&quot;] with 6 slots
##   .. .. ..@ start          : int [1:20] 27852 764822 399 4317479 6701031 669785 3756431 1809654 1390917 1013851 ...
##   .. .. ..@ width          : int [1:20] 2338 2357 2357 1801 2146 2156 2373 2364 2358 2529 ...
##   .. .. ..@ NAMES          : NULL
##   .. .. ..@ elementType    : chr &quot;ANY&quot;
##   .. .. ..@ elementMetadata: NULL
##   .. .. ..@ metadata       : list()
##   ..@ strand         :Formal class 'Rle' [package &quot;S4Vectors&quot;] with 4 slots
##   .. .. ..@ values         : Factor w/ 3 levels &quot;+&quot;,&quot;-&quot;,&quot;*&quot;: 2 1 2 1 2 1 2
##   .. .. ..@ lengths        : int [1:7] 1 2 3 2 5 3 4
##   .. .. ..@ elementMetadata: NULL
##   .. .. ..@ metadata       : list()
##   ..@ seqinfo        :Formal class 'Seqinfo' [package &quot;GenomeInfoDb&quot;] with 4 slots
##   .. .. ..@ seqnames   : chr [1:20] &quot;ZYGRO02696&quot; &quot;ASHGO01343&quot; &quot;CANAL00775&quot; &quot;CANAW00573&quot; ...
##   .. .. ..@ seqlengths : int [1:20] NA NA NA NA NA NA NA NA NA NA ...
##   .. .. ..@ is_circular: logi [1:20] NA NA NA NA NA NA ...
##   .. .. ..@ genome     : chr [1:20] NA NA NA NA ...
##   ..@ elementMetadata:Formal class 'DataFrame' [package &quot;S4Vectors&quot;] with 6 slots
##   .. .. ..@ rownames       : NULL
##   .. .. ..@ nrows          : int 20
##   .. .. ..@ listData       : Named list()
##   .. .. ..@ elementType    : chr &quot;ANY&quot;
##   .. .. ..@ elementMetadata: NULL
##   .. .. ..@ metadata       : list()
##   ..@ elementType    : chr &quot;ANY&quot;
##   ..@ metadata       : list()</code></pre>
<p>The user can also obtain the list of sequences for a given dataframe of proteins and well as the list of their corresponding ontologies (that can be plugged into the topGO for further analysis). This can be done using the functions getSequences() and getOntologies() respectively.</p>
<p>For further information on the OMA REST API please visit <a href="http://omadev.cs.ucl.ac.uk/api/docs">OMA REST API DOCUMENTATION</a>.</p>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
